prime-router/src/main/kotlin/azure/GetSubmissions.kt

```diff
@@ -0,0 +1,72 @@
+package gov.cdc.prime.router.azure
+
+import com.microsoft.azure.functions.HttpMethod
+import com.microsoft.azure.functions.HttpRequestMessage
+import com.microsoft.azure.functions.HttpResponseMessage
+import com.microsoft.azure.functions.annotation.AuthorizationLevel
+import com.microsoft.azure.functions.annotation.FunctionName
+import com.microsoft.azure.functions.annotation.HttpTrigger
+import gov.cdc.prime.router.tokens.OktaAuthentication
+import org.apache.logging.log4j.kotlin.logger
+
+/**
+ * Submissions API
+ * Returns a list of Actions from `public.action`.
+ */
+
+class GetSubmissions(
+    submissionsFacade: SubmissionsFacade = SubmissionsFacade.common,
+    oktaAuthentication: OktaAuthentication = OktaAuthentication(PrincipalLevel.SYSTEM_ADMIN)
+) {
+    private val facade = submissionsFacade
+    private val oktaAuthentication = oktaAuthentication
+
+    @FunctionName("getSubmissions")
+    fun run(
+        @HttpTrigger(
+            name = "getSubmissions",
+            methods = [HttpMethod.GET],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "submissions"
+        ) request: HttpRequestMessage<String?>,
+    ): HttpResponseMessage {
+        val qLimit = request.queryParameters["limit"]
+
+        var limit = 10
+        if (!qLimit.isNullOrEmpty()) {
+            if (isNumber(qLimit))
+                limit = qLimit.toInt()
+            else
+                return HttpUtilities.bad(request, "Limit must be an integer.")
+        }
+
+        return when (request.httpMethod) {
+            HttpMethod.GET -> getList(request, limit)
```

Comment:
Should be able to configure the method in the `@HttpTrigger`

Reply:
Do you have an example of this elsewhere? On my looking through the codebase this seems to be a common pattern.

Reply:
Yup there are, [here's one from settings](https://github.com/CDCgov/prime-reportstream/blob/master/prime-router/src/main/kotlin/azure/SettingsFunctions.kt#L27)